### DISCUSSION TITLE
Routes work strange

### DISCUSSION THREAD

**Discussion by onderbakirtas** [sveltejs/kit#2787] [Q&A]
Hello, all. I think this is not documented or I couldn't find in the docs. Say that we have 2 files, one is named `routes/[slug].svelte` and the other one is `routes/a.svelte`. When I navigate to `abc.com/a`, I expect to see the contents of the `a.svelte` file, but instead `[slug].svelte` file takes the priority show its contents. Is this desired behavior? Same thing happens when I place `a.svelte` in its own folder, like this `a/index.svelte`. Thanks for your answers.

**Comment by geoffrich** [comment]
I can't seem to repro this -- do you have an example repo? I created `[slug].svelte` and `a.svelte` in `src/routes` as below. Navigating to `http://localhost:3001/a` shows me the `a.svelte` component, as you wanted. Navigating to `http://localhost:3001/something` shows me the slug component.

```html
<!-- a.svelte -->
<h1>Hello, I'm A!</h1>
```

```html
<!-- [slug].svelte -->
<script>
	import { page } from '$app/stores';
</script>

<h1>This is a slug: {$page.params.slug}</h1>
```

The docs on [fallthrough routes](https://kit.svelte.dev/docs#routing-advanced-fallthrough-routes) might be relevant, though I think it would indicate that `/a` would be returned because it's the best match.
